src/tools/wasm-split/instrumenter.cpp

```diff
@@ -62,6 +69,22 @@ void Instrumenter::addGlobals() {
   }
 }
 
+void Instrumenter::addSecondaryMemory(size_t profileSize) {
+  if (options.storageKind != WasmSplitOptions::StorageKind::InSecondaryMemory) {
+    // Don't need secondary memory
+    return;
+  }
+  if (!wasm->features.hasMultiMemories()) {
+    Fatal()
+      << "error: --in-secondary-memory requires multi-memories to be enabled";
+  }
+
+  secondaryMemory = Names::getValidMemoryName(*wasm, "split_data");
+  // Create a memory with enough pages to write into
+  size_t pages = (profileSize + Memory::kPageSize - 1) / Memory::kPageSize;
```

Comment:
I could change the pages to: (numFuncs + Memory::kPageSize - 1) / Memory::kPageSize.